webapp/src/ts/services/telemetry.service.ts

```diff
@@ -212,10 +212,23 @@ export class TelemetryService {
   }
 
   private async getCurrentTelemetryDB(today: TodayMoment, telemetryDBs) {
```

Comment:
Looking at this closer, I think we should be able to wildly simplify this function.  `generateTelemetryDBName` should always return the expected name for the given date/user.  So, there is not really any need to check the current `telemetryDBs` values. We can just directly return the result of `generateTelemetryDBName`.

```ts
  private async getCurrentTelemetryDB(today: TodayMoment) {
  const currentDB = this.generateTelemetryDBName(today);
  return this.windowRef.PouchDB(currentDB); // Avoid angular-pouch as digest isn't necessary here
}
```